### ISSUE TITLE
Question: Why use label smoothed trainer output

### ISSUE THREAD

**Issue by bartmch** [facebookresearch/deit#56]
More of a methodological question, then a repo-related question.
Wouldn't it make more sense to use softmax teacher output to train the student? As opposed to using the label smoothed teacher (hard) output? Why did you choose the label smoothing step if you have access to the teacher model (ie. it's logits or it's softmax output)?

Thanks and great work!

**Comment by TouvronHugo** [comment]
Hi @bartmch,

Excellent remark/question,

We have tried both. The hard distillation that we propose in our paper works better. 

This may be due to the fact that, with hard distillation, we have a training that is closer to the training without distillation, and therefore there is no need to adapt the hyper-params when moving from training with labels to training with a teacher. 

But we are not even if the hyper-parameter is actually the best explanation: we have tried some adaptation of the soft distillation, in particular by adapting 
* the temperature of the softmax
* the learning rate
* the weight \lambda between the label and the teacher prediction,
but we could not get significantly better results than what we report in the paper with traditional soft distillation. 

Last remark: for the teacher branch, note that we don’t label smoothing (the smoothing is implicit to the data-augmentation that can provide different labels for the same image).

Best,

Hugo
